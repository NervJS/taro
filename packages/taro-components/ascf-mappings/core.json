{
  "Button": {
    "size": {
      "ascfProp": "dimension",
      "valueMap": {
        "mini": "XS",
        "small": "S",
        "default": "M",
        "large": "L"
      },
      "required": false,
      "default": "M",
      "docLink": "https://ascf.docs/components/button#dimension"
    },
    "loading": {
      "ascfProp": "isLoading",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/button#isloading"
    }
  },
  "Input": {
    "password": {
      "ascfProp": "secretInput",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/input#secretinput"
    }
  }
}
