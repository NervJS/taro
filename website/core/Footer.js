/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

const React = require('react')
class Footer extends React.Component {
  render () {
    return (
      <footer className='footer' id='footer'>
        <div className='grid_c1 footer_cont'>
          <div className='footer_logo_container'>
            <div className='footer_logo' />
            <span className='footer_designedby' />
          </div>
          <div className='footer_link_container'>
            <div className='footer_link'>
              <h3 className='footer_link_tit footer_link_tit1'>相关资源</h3>
              <p><a className='link' href='https://taro.jd.com/' target='_blank'>Taro</a></p>
              <p><a className='link' href='https://taro-ui.jd.com/' target='_blank'>Taro UI</a></p>
              <p><a className='link' href='https://at-ui.github.io/at-ui/#/zh' target='_blank'>At-UI</a></p>
              <p><a className='link' href='https://nerv.aotu.io/' target='_blank'>Nerv</a></p>
              <p><a className='link' href='https://athena.aotu.io/' target='_blank'>Athena</a></p>
            </div>
            <div className='footer_link'>
              <h3 className='footer_link_tit footer_link_tit2'>社区</h3>
              <p><a href='https://github.com/NervJS/taro/issues' target='_blank'>GitHub</a></p>
              <p><a href='https://taro-club.jd.com' target='_blank'>Taro BBS</a></p>
              <p className='footer_link_connect_wrap'>
                <span className='footer_link_connect footer_link_wechat'>微信<span className='wechat_qrcode_icon'>
                  <svg t='1554966525626' className='icon svgicon' viewBox='0 0 1024 1024' version='1.1' xmlns='http://www.w3.org/2000/svg' p-id='2588' data-spm-anchor-id='a313x.7781069.0.i0'>
                    <path d='M240.071 241.095h59.278v59.278h-59.278v-59.278z' fill='' p-id='2589' />
                    <path d='M405.959 134.485h-272.611v272.611h106.723v47.445h59.278v-47.445h106.723v-59.278h47.445v-47.445h-47.445l-0.114-165.888zM346.795 347.819h-154.169v-154.055h154.055v154.055h0.114zM240.071 727.154h59.278v59.278h-59.278v-59.278zM726.016 241.095h59.278v59.278h-59.278v-59.278zM512.683 509.042v63.943h59.278v-59.165h47.445v-59.278h-47.445v-47.445h-59.278v101.945zM512.683 725.789v60.643h59.278v-106.723h47.445v-59.278h-106.723v105.358zM571.961 786.432h47.445v47.445h-47.445v-47.445zM453.405 833.877v59.165h118.557v-59.165h-118.557z' fill='' p-id='2590' />
                    <path d='M678.685 679.709h-59.278v106.723h106.61v-59.278h-47.331v-47.445zM726.016 893.042h166.002v-59.165h-106.723v-47.445h-59.278v106.61zM892.018 513.821v-59.278h-106.723v59.278h106.723zM832.739 727.154h59.278v59.278h-59.278v-59.278zM453.405 347.819h59.278v59.278h-59.278v-59.278zM726.016 454.542v-47.445h166.002v-272.611h-272.611v59.278h-106.723v47.445h106.723v59.165h-47.445v47.445h47.445v59.278h59.278v47.445h47.331zM678.685 193.763h154.055v154.055h-154.055v-154.055zM678.685 572.985h47.331v47.445h-47.331v-47.445zM785.294 679.709h-59.278v47.445h106.723v-106.723h59.278v-47.445h-106.723v106.723zM453.405 241.095h59.278v59.278h-59.278v-59.278zM299.349 513.821h47.445v59.165h-47.445v-59.165zM453.405 454.542h-106.61v59.278h59.165v59.165h47.445v-118.443z' fill='' p-id='2591' />
                    <path d='M405.959 786.432v-106.723h47.445v-59.278h-213.333v-106.61h-106.723v59.278h59.278v47.445h-59.278v272.611h272.611v-59.278h47.445v-47.445h-47.445zM346.795 833.877h-154.169v-154.169h154.055v154.169h0.114zM453.405 572.985h59.278v47.445h-59.278v-47.445zM619.406 513.821h59.278v59.165h-59.278v-59.165zM726.016 513.821h59.278v59.165h-59.278v-59.165z' fill='' p-id='2592' />
                  </svg>
                </span></span>
                <span className='footer_link_wechat_img'>
                  <img src='https://img11.360buyimg.com/ling/jfs/t1/105094/21/15729/35411/5e74e903Ed1490359/5991e49a33964c93.png' />
                </span>
              </p>
            </div>
            <div className='footer_link'>
              <h3 className='footer_link_tit footer_link_tit3'>关于我们</h3>
              <p><a href='https://aotu.io/' target='_blank'>凹凸实验室</a></p>
              <p><a href='https://aotu.io/join/' target='_blank'>加入我们</a></p>
              <p><a href='mailto:taro@jd.com?subject=【Taro 合作】合作标题'>联系我们</a></p>
            </div>
            <div className='footer_link'>
              <h3 className='footer_link_tit footer_link_tit4'>感谢</h3>
              <p><a href='http://jdc.jd.com/' target='_blank'>用户体验设计部</a></p>
              <p><a href='https://github.com/nervjs/taro#%E8%B4%A1%E7%8C%AE%E8%80%85%E4%BB%AC' target='_blank'>Taro 贡献者们</a></p>
            </div>
          </div>
        </div>
        <div className='copyright'>
          <div className='in'>Copyright © 2019. All Rights Reserved. 粤ICP备15077732号-2</div>
        </div>
      </footer>
    )
  }
}

module.exports = Footer
