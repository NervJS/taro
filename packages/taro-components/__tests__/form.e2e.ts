import { E2EPage, newE2EPage } from '@stencil/core/testing'

describe('Form e2e', () => {
  let page: E2EPage
  const switchChecked = false
  const sliderValue = 0
  const inputValue = ''
  const textareaValue = ''
  const radioList = [{
    value: 'radio1',
    checked: false
  }, {
    value: 'radio2',
    checked: false
  }]
  const checkboxList = [{
    value: 'checkbox1',
    checked: false
  }, {
    value: 'checkbox2',
    checked: false
  }]
  const pickerValue = 0

  it('events', async () => {
    page = await newE2EPage({
      html: `<taro-form-core>
        <taro-switch-core name='my-switch' checked="${switchChecked}"></taro-switch-core>
        <taro-slider-core name='my-slider' value="${sliderValue}"></taro-slider-core>
        <taro-input-core name='my-input' value="${inputValue}"></taro-input-core>
        <taro-textarea-core name='my-textarea' value="${textareaValue}"></taro-textarea-core>
        <taro-radio-group-core name='my-radio-group'>
          ${radioList.map(({ value, checked }) => `<taro-radio-core key="${value}" value="${value}" checked="${checked}">${value}</taro-radio-core>`).join('')}
        </taro-radio-group-core>
        <taro-checkbox-group-core name='my-checkbox-group'>
          ${checkboxList.map(({ value, checked }) => `<taro-checkbox-core key="${value}" value="${value}" checked="${checked}">${value}</taro-checkbox-core>`).join('')}
        </taro-checkbox-group-core>
        <taro-picker-core name='my-picker' range={['葡萄', '橙子', '苹果', '木瓜']} value="${pickerValue}"></taro-picker-core>
        <taro-button-core form-type='submit'>Submit</taro-button-core>
        <taro-button-core form-type='reset'>Reset</taro-button-core>
      </taro-form-core>`,
    })
    const el = await page.find('taro-form-core')
    const submit = await el.find('taro-button-core[form-type="submit"]')
    const reset = await el.find('taro-button-core[form-type="reset"]')
    const formSwitch = await el.find('taro-switch-core')
    const formSlider = await el.find('taro-slider-core')
    const formInput = await el.find('taro-input-core')
    const formTextarea = await el.find('taro-textarea-core')
    const formRadio = await el.find('taro-radio-group-core')
    const formCheckbox = await el.find('taro-checkbox-group-core')
    const formPicker = await el.find('taro-picker-core')
    const onSubmit = await el.spyOnEvent('submit')
    const onReset = await el.spyOnEvent('reset')

    submit.triggerEvent('touchend')
    await page.waitForChanges()
    expect(onSubmit).toHaveReceivedEventTimes(1)
    expect(onSubmit).toHaveReceivedEventDetail({
      value: {
        'my-switch': false,
        'my-slider': '0',
        'my-input': '',
        'my-textarea': '',
        'my-radio-group': '',
        'my-checkbox-group': [],
        'my-picker': '0'
      }
    })

    formSwitch.setProperty('checked', true)
    formSlider.setProperty('value', '60')
    formInput.setProperty('value', 'taro-input')
    formTextarea.setProperty('value', 'taro-textarea')
    formPicker.setProperty('value', '1')
    await page.waitForChanges()
    ;(await formCheckbox.find('taro-checkbox-core[key="checkbox1"]')).click()
    await page.waitForChanges()
    ;(await formCheckbox.find('taro-checkbox-core[key="checkbox2"]')).click()
    await page.waitForChanges()
    ;(await formRadio.find('taro-radio-core[key="radio1"]')).click()
    await page.waitForChanges()

    submit.triggerEvent('touchend')
    await page.waitForChanges()
    expect(onSubmit).toHaveReceivedEventTimes(2)
    expect(onSubmit).toHaveReceivedEventDetail({
      value: {
        'my-switch': true,
        'my-slider': '60',
        'my-input': 'taro-input',
        'my-textarea': 'taro-textarea',
        'my-radio-group': 'radio1',
        'my-checkbox-group': ['checkbox1', 'checkbox2'],
        'my-picker': '1'
      }
    })

    reset.triggerEvent('touchend')
    await page.waitForChanges()
    expect(onReset).toHaveReceivedEventTimes(1)

    submit.triggerEvent('touchend')
    await page.waitForChanges()
    expect(onSubmit).toHaveReceivedEventDetail({
      value: {
        'my-switch': false,
        'my-input': '',
        'my-textarea': '',
        'my-radio-group': '',
        'my-checkbox-group': [],
        // Slider 和 Picker 的 input type 为 hidden，form.reset() 不能重置它们，需要再想想办法
        'my-slider': '60',
        'my-picker': '1'
      }
    })

    expect(page).toMatchSnapshot()
  })
})
