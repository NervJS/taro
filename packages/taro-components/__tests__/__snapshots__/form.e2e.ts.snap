// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form e2e events 1`] = `
CDPPage {
  "_e2eClose": [Function],
  "_e2eElements": Array [
    <taro-form-core class="hydrated">
  <!---->
  <form>
    <taro-switch-core checked="false" class="hydrated" name="my-switch">
      <input class="weui-switch" name="my-switch" type="checkbox" style="border-color: rgb(4, 190, 2); background-color: rgb(4, 190, 2);">
    </taro-switch-core>
    <taro-slider-core class="hydrated weui-slider-box" name="my-slider" value="60">
      <div class="weui-slider">
        <div class="weui-slider__inner" style="background-color: rgb(233, 233, 233);">
          <div class="weui-slider__track" style="width: 60%; background-color: rgb(26, 173, 25);"></div>
          <div class="weui-slider__handler" style="left: 60%; width: 28px; height: 28px; background-color: rgb(255, 255, 255); margin-top: -14px; margin-left: -14px;"></div>
          <input name="my-slider" type="hidden" value="60">
        </div>
      </div>
    </taro-slider-core>
    <taro-input-core class="hydrated" name="my-input" value="">
      <input class="weui-input" maxlength="140" name="my-input" type="text">
    </taro-input-core>
    <taro-textarea-core class="hydrated" name="my-textarea" value=""><textarea class="taro-textarea" maxlength="140" name="my-textarea"></textarea>
    </taro-textarea-core>
    <taro-radio-group-core class="hydrated weui-cells_radiogroup" name="my-radio-group">
      <taro-radio-core checked="" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="radio1" name="my-radio-group" value="radio1">
        <!---->
        <input class="weui-check" name="my-radio-group" type="radio" value="radio1">
        <i class="weui-icon-checked"></i>
        radio1
      </taro-radio-core>
      <taro-radio-core checked="false" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="radio2" name="my-radio-group" value="radio2">
        <!---->
        <input class="weui-check" name="my-radio-group" type="radio" value="radio2">
        <i class="weui-icon-checked"></i>
        radio2
      </taro-radio-core>
    </taro-radio-group-core>
    <taro-checkbox-group-core class="hydrated" name="my-checkbox-group">
      <taro-checkbox-core checked="false" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="checkbox1" name="my-checkbox-group" value="checkbox1">
        <!---->
        <input class="taro-checkbox_checked" name="my-checkbox-group" type="checkbox" value="checkbox1">
        checkbox1
      </taro-checkbox-core>
      <taro-checkbox-core checked="false" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="checkbox2" name="my-checkbox-group" value="checkbox2">
        <!---->
        <input class="taro-checkbox_checked" name="my-checkbox-group" type="checkbox" value="checkbox2">
        checkbox2
      </taro-checkbox-core>
    </taro-checkbox-group-core>
    <taro-picker-core 'æ©™å­',="" 'æœ¨ç“œ']}="" 'è‹¹æžœ',="" class="hydrated" name="my-picker" range="{['è‘¡è„'," value="0">
      <!---->
      <div>
        <input name="my-picker" type="hidden" value="1">
      </div>
    </taro-picker-core>
    <taro-button-core class="hydrated" form-type="submit" type="">
      <!---->
      Submit
    </taro-button-core>
    <taro-button-core class="hydrated" form-type="reset" type="">
      <!---->
      Reset
    </taro-button-core>
  </form>
</taro-form-core>,
    <taro-button-core class="hydrated" form-type="submit" type="">
  <!---->
  Submit
</taro-button-core>,
    <taro-button-core class="hydrated" form-type="reset" type="">
  <!---->
  Reset
</taro-button-core>,
    <taro-switch-core checked="false" class="hydrated" name="my-switch">
  <input class="weui-switch" name="my-switch" type="checkbox" style="border-color: rgb(4, 190, 2); background-color: rgb(4, 190, 2);">
</taro-switch-core>,
    <taro-slider-core class="hydrated weui-slider-box" name="my-slider" value="60">
  <div class="weui-slider">
    <div class="weui-slider__inner" style="background-color: rgb(233, 233, 233);">
      <div class="weui-slider__track" style="width: 60%; background-color: rgb(26, 173, 25);"></div>
      <div class="weui-slider__handler" style="left: 60%; width: 28px; height: 28px; background-color: rgb(255, 255, 255); margin-top: -14px; margin-left: -14px;"></div>
      <input name="my-slider" type="hidden" value="60">
    </div>
  </div>
</taro-slider-core>,
    <taro-input-core class="hydrated" name="my-input" value="">
  <input class="weui-input" maxlength="140" name="my-input" type="text">
</taro-input-core>,
    <taro-textarea-core class="hydrated" name="my-textarea" value=""><textarea class="taro-textarea" maxlength="140" name="my-textarea"></textarea>
</taro-textarea-core>,
    <taro-radio-group-core class="hydrated weui-cells_radiogroup" name="my-radio-group">
  <taro-radio-core checked="" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="radio1" name="my-radio-group" value="radio1">
    <!---->
    <input class="weui-check" name="my-radio-group" type="radio" value="radio1">
    <i class="weui-icon-checked"></i>
    radio1
  </taro-radio-core>
  <taro-radio-core checked="false" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="radio2" name="my-radio-group" value="radio2">
    <!---->
    <input class="weui-check" name="my-radio-group" type="radio" value="radio2">
    <i class="weui-icon-checked"></i>
    radio2
  </taro-radio-core>
</taro-radio-group-core>,
    <taro-checkbox-group-core class="hydrated" name="my-checkbox-group">
  <taro-checkbox-core checked="false" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="checkbox1" name="my-checkbox-group" value="checkbox1">
    <!---->
    <input class="taro-checkbox_checked" name="my-checkbox-group" type="checkbox" value="checkbox1">
    checkbox1
  </taro-checkbox-core>
  <taro-checkbox-core checked="false" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="checkbox2" name="my-checkbox-group" value="checkbox2">
    <!---->
    <input class="taro-checkbox_checked" name="my-checkbox-group" type="checkbox" value="checkbox2">
    checkbox2
  </taro-checkbox-core>
</taro-checkbox-group-core>,
    <taro-picker-core 'æ©™å­',="" 'æœ¨ç“œ']}="" 'è‹¹æžœ',="" class="hydrated" name="my-picker" range="{['è‘¡è„'," value="0">
  <!---->
  <div>
    <input name="my-picker" type="hidden" value="1">
  </div>
</taro-picker-core>,
    <taro-checkbox-core checked="false" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="checkbox1" name="my-checkbox-group" value="checkbox1">
  <!---->
  <input class="taro-checkbox_checked" name="my-checkbox-group" type="checkbox" value="checkbox1">
  checkbox1
</taro-checkbox-core>,
    <taro-checkbox-core checked="false" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="checkbox2" name="my-checkbox-group" value="checkbox2">
  <!---->
  <input class="taro-checkbox_checked" name="my-checkbox-group" type="checkbox" value="checkbox2">
  checkbox2
</taro-checkbox-core>,
    <taro-radio-core checked="" class="hydrated weui-cells_checkbox" classname="weui-cells_checkbox" key="radio1" name="my-radio-group" value="radio1">
  <!---->
  <input class="weui-check" name="my-radio-group" type="radio" value="radio1">
  <i class="weui-icon-checked"></i>
  radio1
</taro-radio-core>,
  ],
  "_e2eEventIds": 2,
  "_e2eEvents": Map {
    0 => Object {
      "callback": [Function],
      "eventName": "submit",
    },
    1 => Object {
      "callback": [Function],
      "eventName": "reset",
    },
  },
  "_e2eGoto": [Function],
  "close": [Function],
  "compareScreenshot": [Function],
  "debugger": [Function],
  "emitter": Object {
    "all": Map {
      "console" => Array [
        [Function],
      ],
      "pageerror" => Array [
        [Function],
      ],
      "requestfailed" => Array [
        [Function],
      ],
      "request" => Array [
        [Function],
      ],
    },
    "emit": [Function],
    "off": [Function],
    "on": [Function],
  },
  "eventsMap": Map {
    "console" => Array [
      [Function],
    ],
    "pageerror" => Array [
      [Function],
    ],
    "requestfailed" => Array [
      [Function],
    ],
    "request" => Array [
      [Function],
    ],
  },
  "find": [Function],
  "findAll": [Function],
  "getDiagnostics": [Function],
  "spyOnEvent": [Function],
  "waitForChanges": [Function],
  "waitForEvent": [Function],
}
`;
